Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -90,132 +89,294 @@ namespace ROS2
-        if (!AZStd::regex_search(link_name, match, wheel_regex))
+        if (!AZStd::regex_search(linkName, match, wheelRegex))
         {
             return false;
         }
         // The name should contain a joint word
-        if (AZStd::regex_search(link_name, match, joint_regex))
+        if (AZStd::regex_search(linkName, match, jointRegex))
         {
             return false;
         }
         // Wheels need to have collision and visuals
-        if (!(link->collision && link->visual))
+        if ((link->CollisionCount() == 0) || (link->VisualCount() == 0))
         {
             return false;
         }
-        // Parent joint needs to be CONTINOUS
-        if (link->parent_joint && link->parent_joint->type == urdf::Joint::CONTINUOUS)
+
+        // When this link is a child, the parent link joint needs to be CONTINUOUS
+        AZStd::vector<const sdf::Joint*> joints = GetJointsForChildLink(model,
+            linkName, true);
+
+        // URDFs only have a single parent
+        // This is explained in the Pose frame semantics tutorial for sdformat
+        // http://sdformat.org/tutorials?tut=pose_frame_semantics&ver=1.5#parent-frames-in-urdf
+
+        // The SDF URDF parser converts continuous joints to revolute joints with a limit
+        // of -infinity to +infinity
+        // https://github.com/gazebosim/sdformat/blob/sdf13/src/parser_urdf.cc#L3009-L3039
+        bool isWheel{};
+        if (!joints.empty())
         {
-            return true;
+            const sdf::Joint* potentialWheelJoint = joints.front();
+            using LimitType = decltype(potentialWheelJoint->Axis()->Lower());
+            // There should only be 1 element for URDF, however that will not be verified
+            // in case this function is called on link from an SDF file
+            isWheel = potentialWheelJoint->Type() == sdf::JointType::CONTINUOUS;
+            isWheel = isWheel || (potentialWheelJoint->Type() == sdf::JointType::REVOLUTE
+                && potentialWheelJoint->Axis()->Lower() == -AZStd::numeric_limits<LimitType>::infinity()
+                && potentialWheelJoint->Axis()->Upper() == AZStd::numeric_limits<LimitType>::infinity());
         }
-        return false;
+
+        return isWheel;
     }
 
-    AZ::Transform Utils::GetWorldTransformURDF(const urdf::LinkSharedPtr& link, AZ::Transform t)
+    AZ::Transform GetWorldTransformURDF(const sdf::Link* link, AZ::Transform t)
     {
-        if (link->getParent() != nullptr)
+        // Determine if the pose is relative to another link
+        // See doxygen at http://osrf-distributions.s3.amazonaws.com/sdformat/api/13.2.0/classsdf_1_1SDF__VERSION__NAMESPACE_1_1Link.html#a011d84b31f584938d89ac6b8c8a09eb3
+
+        sdf::SemanticPose linkSemanticPos = link->SemanticPose();
+        gz::math::Pose3d resolvedPose;
+
+        if (sdf::Errors poseResolveErrors = linkSemanticPos.Resolve(resolvedPose);
+            !poseResolveErrors.empty())
```

Comment:
I believe clang-format is combining the lines together since there are space for both the if `initializer` and the `condition` to fit under [140 characters](https://github.com/o3de/o3de/blob/117611cb9c497705a12c26447193da25dec02440/.clang-format#L32)